File: 507-PerfectNumber/507-PerfectNumber.js
// Last updated: 24/04/2026, 10:05:15
/**
 * @param {number} num
 * @return {boolean}
 */
function checkPerfectNumber(num) {
    if (num <= 1) return false;

    let sum = 0;

    for (let i = 1; i <= num / 2; i++) {
        if (num % i === 0) {
            sum += i;
        }
    }

    return sum === num;
}

File: 507-PerfectNumber/507-PerfectNumber.cpp
// Last updated: 05/05/2026, 18:13:16
1class Solution {
2public:
3    bool checkPerfectNumber(int num) {
4        int sum = 0;
5        for(int i = 1; i<num; i++){
6            if(num%i==0){
7                sum += i;
8            }
9        }
10
11        if(sum == num){
12            return true;
13        }
14
15        return false;
16    }
17};